### PR TITLE
IntelGPU backend: SigmoidBackprop operation

### PR DESCRIPTION
**Pull request by aalbersk**

### CODE REVIEW

File: src/ngraph/runtime/intelgpu/intelgpu_op_custom_kernels.cpp [hunk 1]
```diff
@@ -1035,3 +1035,44 @@ void runtime::intelgpu::do_convert_operation(cldnn::topology& topology,
...
+                                                      const Shape& delta_shape,
+                                                      const string& output_name,
+                                                      const Shape& output_shape,
+                                                      const element::Type& output_type)
+{
+    const string entry_point_name = "op_sigmoid_backprop_" + output_name;
+    codegen::CodeWriter writer;
+    vector<size_t> gws;
+
+    runtime::intelgpu::gen_func_def(
```
Comment (shssf) [inline src/ngraph/runtime/intelgpu/intelgpu_op_custom_kernels.cpp]: No need to use `runtime::intelgpu::` in case of the function is inside the namespace.

File: src/ngraph/runtime/intelgpu/intelgpu_op_custom_kernels.cpp [hunk 2]
```diff
@@ -1035,3 +1035,44 @@ void runtime::intelgpu::do_convert_operation(cldnn::topology& topology,
...
+    writer.block_begin();
+    {
+        writer << "float func_x = 0.0f;\n";
+        gws = runtime::intelgpu::generate_loops(writer, output_shape, true);
+
+        writer << "func_x = 1/(1+ exp(-input0" << access_dims(input_shape) << "));\n";
+        writer << "output" << access_dims(output_shape) << " = input1" << access_dims(delta_shape)
+               << " * func_x * (1 - func_x);\n";
+
+        runtime::intelgpu::generate_loops(writer, output_shape, false);
```
Comment (shssf) [inline src/ngraph/runtime/intelgpu/intelgpu_op_custom_kernels.cpp]: The same with namespace as above

File: src/ngraph/runtime/intelgpu/intelgpu_op_custom_kernels.cpp [hunk 3]
```diff
@@ -1035,3 +1035,44 @@ void runtime::intelgpu::do_convert_operation(cldnn::topology& topology,
...
+
+    runtime::intelgpu::gen_func_def(
+        writer, entry_point_name, {2, "float"}, {input_shape, delta_shape}, "float", output_shape);
+
+    writer.block_begin();
+    {
+        writer << "float func_x = 0.0f;\n";
+        gws = runtime::intelgpu::generate_loops(writer, output_shape, true);
+
+        writer << "func_x = 1/(1+ exp(-input0" << access_dims(input_shape) << "));\n";
```
Comment (shssf) [inline src/ngraph/runtime/intelgpu/intelgpu_op_custom_kernels.cpp]: Since we use `float` here, please use `1.0f`. Also, it might be better to use internal compiler macro if it exists.

**Comment by shssf** [comment]
@aalbersk I believe you forgot to do `./maint/apply-code-style.sh`

File: src/ngraph/runtime/intelgpu/intelgpu_op_custom_kernels.cpp [hunk 4]
```diff
@@ -204,18 +204,17 @@ void runtime::intelgpu::do_pad_operation(cldnn::topology& topology,
     vector<size_t> gws;
 
     // The kernel name and parameters
-    runtime::intelgpu::gen_func_def(
-        writer, entry_point_name, {2, "float"}, {input_shape, {1}}, "float", output_shape);
+    gen_func_def(writer, entry_point_name, {2, "float"}, {input_shape, {1}}, "float", output_shape);
```
Comment (rkimballn1) [inline src/ngraph/runtime/intelgpu/intelgpu_op_custom_kernels.cpp]: It is better to do refactoring like this, removing namespace, in a separate PR. All this does is add deltas that are unrelated to this PR.
